### ISSUE TITLE
[Bug] [zos_copy] Encoding normalization applies to binary files

### ISSUE THREAD

**Issue by rexemin** [ansible-collections/ibm_zos_core#807]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.2

### IBM Enterprise Python

v3.10.x

### IBM z/OS Ansible core Version

v1.5.0

### ansible-version

v2.14.x

### z/OS version

v2.3

### Ansible module

zos_copy

### Bug description

When copying a binary file from USS, the code that will normalize encoding for USS files gets executed. This code should only apply to text files, never to binary ones, so the module terminates with an error:
```json
{"changed": false, "msg": "An error occurred during encoding: \"iconv: Conversion from codeset \"binary\" to \"UTF-8\": EDC5121I Invalid argument.\n\"", "rc": null, "stderr": null, "stderr_lines": null, "stdout": null, "stdout_lines": null}
```

This error happens if the source file is tagged as `binary` and even when the task specifies `is_binary: true`.

**Comment by ddimatos** [comment]
@rexemin , the branch for 1.6 is `stagging-v1.6.0`, thank you.

**Comment by ddimatos** [comment]
@rexemin - since this is in 1.6, when we release and perform the upstream merge into our dev branch, this should automatically be included, thus is also needed in our `dev` (`1.7.0-beta.1`) but should happen organically.
